kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/engine/DefaultJsonHandlerSuite.scala

```diff
@@ -373,6 +373,65 @@ class DefaultJsonHandlerSuite extends AnyFunSuite with TestUtils with DefaultVec
     checkAnswer(actResult, expResult)
   }
 
+  test("read json files, some of which are empty") {
+    val testFiles = fsClient.listFrom(getTestResourceFilePath("json-files-with-empty/1.json"))
+    val actResult = jsonHandler.readJsonFiles(
+      testFiles,
+      new StructType()
+        .add("path", StringType.STRING)
+        .add("size", LongType.LONG)
+        .add("dataChange", BooleanType.BOOLEAN),
+      Optional.empty()
+    ).toSeq.map(batch => TestRow(batch.getRows.next))
+
+    val expResult = Seq(
+      TestRow("part-00000-d83dafd8-c344-49f0-ab1c-acd944e32493-c000.snappy.parquet", 348L, true),
+      TestRow("part-00000-cb078bc1-0aeb-46ed-9cf8-74a843b32c8c-c000.snappy.parquet", 687L, true),
+      TestRow("part-00001-9bf4b8f8-1b95-411b-bf10-28dc03aa9d2f-c000.snappy.parquet", 705L, true),
+      TestRow("part-00000-0441e99a-c421-400e-83a1-212aa6c84c73-c000.snappy.parquet", 650L, true),
+      TestRow("part-00001-34c8c673-3f44-4fa7-b94e-07357ec28a7d-c000.snappy.parquet", 650L, true),
+      TestRow("part-00000-842017c2-3e02-44b5-a3d6-5b9ae1745045-c000.snappy.parquet", 649L, true),
+      TestRow("part-00001-e62ca5a1-923c-4ee6-998b-c61d1cfb0b1c-c000.snappy.parquet", 649L, true)
+    )
+
+    checkAnswer(actResult, expResult)
+  }
+
+  test("read json files from middle of directory, some of which are empty") {
+    val testFiles = fsClient.listFrom(getTestResourceFilePath("json-files-with-empty/5.json"))
+    val actResult = jsonHandler.readJsonFiles(
+      testFiles,
+      new StructType()
+        .add("path", StringType.STRING)
+        .add("size", LongType.LONG)
+        .add("dataChange", BooleanType.BOOLEAN),
+      Optional.empty()
+    ).toSeq.map(batch => TestRow(batch.getRows.next))
```

Comment:
seems like this code is common to all test. may be arrange the tests as follows:

```
Seq(
  (
     ..., // test files
     expectedResults
  ),
  (
   ...
  )
).foreach {
   case (testfiles, expResults) => 
      // test code.
}
   
```